contracts/peg/curve/CurveRouter.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import {ICurveRouter} from "./ICurveRouter.sol";
+import {ICurvePool} from "./ICurvePool.sol";
+import {IVolt} from "../../volt/IVolt.sol";
+import {IPegStabilityModule} from "../IPegStabilityModule.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "hardhat/console.sol";
+
+import {Constants} from "../../Constants.sol";
+
+contract CurveRouter is ICurveRouter {
+    using SafeERC20 for IERC20;
+
+    /// @notice reference to the Volt contract used.
+    /// Router can be redeployed if Volt address changes
+    IVolt public immutable override volt;
+
+    constructor(IVolt _volt) {
+        volt = _volt;
```

Comment:
constructor should approve the PSM to spend the router's stablecoins

Reply:
also should approve each curve pool's address to spend all pair stables so you don't have to do another approval inside the mint function